service/controller/resource/certconfig/desired.go

```diff
@@ -173,6 +195,48 @@ func (r *Resource) newSpecForEtcd(cc controllercontext.Context, cr apiv1alpha2.C
 	}
 }
 
+func (r *Resource) newSpecForEtcd1(cc controllercontext.Context, cr apiv1alpha2.Cluster) corev1alpha1.CertConfigSpecCert {
```

Comment:
I know it is a bit ugly but I found the alternative worse when looking at `EnsureCreated`. Going with the straight forward approach here.

Reply:
Yeah, I think this is perfectly acceptable for now. Easy to understand and if there's a way, then refactor it out later.

Reply:
It would still be nicer not to duplicate this function three times and just pass the number to get the correct cert, wouldn't it? Am I missing something?

Reply:
I did not like it either. When you think this is better you can gladly create a PR and change it. No real strong feelings here.

Reply:
Sure, no blocker from my side.